### ISSUE TITLE
Questions about diffusion model

### ISSUE THREAD

**Issue by FrederikWarburg** [nianticlabs/diffusionerf#2]
Thanks for the cool paper. I'm trying to reimplement the local patch-based diffusion model. I have some questions

1) I assume that you concatenate the input RGBD? 

2) How do you normalize the depth from hypersim? The depth is in meters? and I assume you somehow normalize to the range [-1,1], a similar range as the RGB values? Also, how do you normalize the depth from the rendering nerfs, such that it is in the range [-1, 1]?

3) What scheduler (DDIM? DDPM?) do you for the reverse process? Figure 3 c?

4) What architecture do you use for the diffusion model? I'm currently implementing it in huggingface like this (bottleneck resolution 6x6).

```
        model = UNet2DModel(
            sample_size=48,  # the target image resolution
            in_channels=len(config.input),  # the number of input channels, 3 for RGB images
            out_channels=len(config.input),  # the number of output channels
            layers_per_block=2,  # how many ResNet layers to use per UNet block
            block_out_channels=(64, 128, 256, 512),
            down_block_types=(
                "DownBlock2D",
                "DownBlock2D",
                "AttnDownBlock2D",
                "DownBlock2D",
            ),
            up_block_types=(
                "UpBlock2D",
                "AttnUpBlock2D",
                "UpBlock2D",
                "UpBlock2D",
            ),
        )
```

**Comment by jamiewynn** [comment]
Hi there, thanks for your interest in our paper.

1. Yes that's right, RGB and D are concatenated in one dense input tensor.
2. We normalize the depth (which, as you say, is in meters with Hypersim) by clipping it so it is at least equal to min_depth (a parameter which we set to 0.2 metres), then inverting it. This gives us an inverse depth in [0, 1], which we then linearly scale to [-1, 1] before feeding it into the model. So to clarify, the D in the RGBD patches is actually this inverse depth. This same procedure is done both in training (when using the Hypersim ground truth depths) and at test time (when rendered depths from the NeRF).
3. We use a cosine beta schedule for the reverse process.
4. If you use [this](https://github.com/lucidrains/denoising-diffusion-pytorch) implementation of the DDPM (which I think is related to the Huggingface one), then the architecture is:
`model = Unet(dim=64, dim_mults=(1, 2, 4, 8), channels=4)`

**Comment by daniyar-niantic** [comment]
Thank you for the interest in the paper!

Please check the code and supplementary pdf, it should have more details. If not, please feel free to open a new issue.
